pkgs/build-support/deno/fetch-deno-deps/default.nix

```diff
@@ -0,0 +1,257 @@
+{
+  callPackage,
+  stdenvNoCC,
+  lib,
+  cacert,
+}:
+let
+  inherit (callPackage ./scripts/deno/default.nix { }) fetch-deno-deps-scripts;
+  inherit (callPackage ./scripts/rust/file-structure-transformer-vendor/default.nix { })
+    file-structure-transformer-vendor
+    ;
+
+  transformDenoLock = lib.extendMkDerivation {
+    constructDrv = stdenvNoCC.mkDerivation;
+    # don't let these bleed into the derivation environment
+    excludeDrvArgNames = [
+      "denoLock"
+    ];
+    extendDrvArgs =
+      finalAttrs:
+      {
+        denoLock,
+        pname,
+        version,
+        ...
+      }@args:
+      {
+        name = args.name or "denoDeps-transformedDenoLock-${finalAttrs.pname}-${finalAttrs.version}";
+        src = args.src or null;
```

Comment:
This is a bit of a weird thing to write as it make `src` conditional, but the buildPhase never uses src. I personally would unconditonally make `src = denoLock`, then it should just do a `cat` unpack.